src/lib.rs

```diff
@@ -22,104 +63,192 @@ impl fmt::Display for DeviceType {
+    }
+
+    /// Returns the maximum color temperature supported by the device in Kelvin.
+    #[must_use]
+    pub fn maximum_temperature_in_kelvin(&self) -> u16 {
+        MAXIMUM_TEMPERATURE_IN_KELVIN
+    }
+}
+
+const VENDOR_ID: u16 = 0x046d;
+const USAGE_PAGE: u16 = 0xff43;
+
+fn device_type_from_product_id(product_id: u16) -> Option<DeviceType> {
+    match product_id {
+        0xc900 => DeviceType::LitraGlow.into(),
+        0xc901 => DeviceType::LitraBeam.into(),
+        0xb901 => DeviceType::LitraBeam.into(),
+        0xc903 => DeviceType::LitraBeamLX.into(),
+        _ => None,
     }
 }
 
 const MINIMUM_TEMPERATURE_IN_KELVIN: u16 = 2700;
 const MAXIMUM_TEMPERATURE_IN_KELVIN: u16 = 6500;
 
-pub fn get_connected_devices<'a>(
-    api: &'a HidApi,
-    serial_number: Option<&'a str>,
-) -> impl Iterator<Item = Device> + 'a {
-    let litra_devices = api
-        .device_list()
-        .filter(|device| {
-            device.vendor_id() == VENDOR_ID
-                && PRODUCT_IDS.contains(&device.product_id())
-                && device.usage_page() == USAGE_PAGE
-        })
-        .filter(move |device| {
-            serial_number.is_none()
-                || serial_number.as_ref().is_some_and(|expected| {
-                    device
-                        .serial_number()
-                        .is_some_and(|actual| &actual == expected)
-                })
-        });
-
-    litra_devices
-        .filter_map(|device| match api.open_path(device.path()) {
-            Ok(device_handle) => Some((device, device_handle)),
-            Err(err) => {
-                println!("Failed to open device {:?}: {:?}", device.path(), err);
-                None
-            }
-        })
-        .map(|(device, device_handle)| {
-            let device_type = get_device_type(device.product_id());
-            let is_on = is_on(&device_handle, &device_type);
-            let brightness_in_lumen = get_brightness_in_lumen(&device_handle, &device_type);
-            let temperature_in_kelvin = get_temperature_in_kelvin(&device_handle, &device_type);
-            let minimum_brightness_in_lumen = get_minimum_brightness_in_lumen(&device_type);
-            let maximum_brightness_in_lumen = get_maximum_brightness_in_lumen(&device_type);
-
-            Device {
-                serial_number: device.serial_number().unwrap_or("").to_string(),
-                device_type,
-                is_on,
-                brightness_in_lumen,
-                temperature_in_kelvin,
-                device_handle,
-                minimum_brightness_in_lumen,
-                maximum_brightness_in_lumen,
-                minimum_temperature_in_kelvin: MINIMUM_TEMPERATURE_IN_KELVIN,
-                maximum_temperature_in_kelvin: MAXIMUM_TEMPERATURE_IN_KELVIN,
-            }
-        })
+/// Returns an [`Iterator`] of connected devices supported by this library.
+pub fn get_connected_devices(api: &HidApi) -> impl Iterator<Item = Device<'_>> {
```

Comment:
One potential risk with this design is that the library's interface seems to be quite tightly coupled to `hidapi`. This is probably an academic concern, but I just wanted to raise it and get your take!

Reply:
I agree. Ideally the library user should not have anything to do with the hidapi transitive dependency. We could add a `Litra` structure that just encapsulates the hidapi object, and make this method a member of that.

Reply:
https://github.com/timrogers/litra-rs/pull/21/commits/a5b67d4723d0d9e1145e9a6f4b1052cba4a30180